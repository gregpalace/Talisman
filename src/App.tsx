import React from 'react';
import { connect } from 'react-redux';

import Footer from './client/components/Footer';
import World from './client/containers/World';
import Viewport from './client/components/hud/ViewPort';

import '../src/stylesheets/index.scss';

const App = ({ appState }: any) => {
  let showFooter = true;
  return (
    <>
      <div className='app-container'>
        <Viewport>
          <World />
        </Viewport>
      </div>
      {showFooter && <Footer />}
    </>
  );
};

const mapStateToProps = ({ appState }: any) => ({ appState });

export default connect(mapStateToProps)(App);
