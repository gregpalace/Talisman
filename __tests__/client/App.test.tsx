import React from 'react';
import ReactDOM from 'react-dom';
import { render, unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../../src/App';

describe('React unit test', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  describe('<App />', () => {
    it('should render', () => {
      expect(wrapper.length).toBe(1);
      expect(wrapper).toBeTruthy();
    });
    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});