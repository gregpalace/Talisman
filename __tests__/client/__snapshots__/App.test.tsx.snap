// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit test <App /> should match snapshot 1`] = `undefined`;
